dataset/util.cpp

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Matthew Andrew
+#include "scipp/dataset/util.h"
+
+namespace scipp {
+SCIPP_DATASET_EXPORT scipp::index size_of(const VariableConstView &view) {
+  auto value_size = view.underlying().data().dtype_size();
+  auto variance_scale = view.hasVariances() ? 2 : 1;
+  return view.dims().volume() * value_size * variance_scale;
```

Comment:
Where is the handling of bucket variables we discussed yesterday?

Reply:
This is currently a work in progress. I'll move the PR out of draft once it it feature complete.